app.py

```diff
@@ -784,10 +785,41 @@ def market_snap(snap_name):
         flask.abort(details_response.status_code, message)
 
     details = details_response.json()
-    metadata = get_metadata_snap(details['snap_id'])
+    metadata = metadata_snap(details['snap_id'])
 
     return flask.render_template(
         'publisher/market.html',
         snap_id=details['snap_id'],
         metadata=metadata
     )
+
+
+@app.route('/account/snaps/<snap_name>/market/', methods=['POST'])
+def post_market_snap(snap_name):
+    if not authentication.is_authenticated(flask.session):
+        return redirect_to_login()
+
+    whitelist = [
+        'title',
+        'summary',
+        'description',
+        'contact_url',
+        'keywords',
+        'license',
+        'price',
+        'blacklist_countries',
+        'whitelist_countries'
+    ]
+
+    body_json = {
+        key: flask.request.form[key]
+        for key in whitelist if key in flask.request.form
+    }
+
+    metadata_snap(flask.request.form['snap_id'], body_json)
+
+    return flask.render_template(
```

Comment:
This may look a bit weird when posting to a form returns back to the same page. But let's see how it works and improve from there.

Reply:
yep for now I added this because I didn't really where to redirect...